### PR TITLE
CAS 8848 implement multi-threading in ClassicalStatistics::_createDataArray()

### PR DESCRIPTION
**Pull request by dmehring**

### CODE REVIEW

File: scimath/Mathematics/ClassicalStatistics.tcc [hunk 1]
```diff
@@ -1125,69 +1127,114 @@ void ClassicalStatistics<CASA_STATP>::_createDataArray(
...
+            nBlocks, extra, nthreads, dataIter,
+            maskIter, weightsIter, offset, nThreadsMax
+        );
+#pragma omp parallel for num_threads(nthreads)
+        for (uInt i=0; i<nBlocks; ++i) {
+#ifdef _OPENMP
+            uInt tid = omp_get_thread_num();
+#else
+            uInt tid = 0;
+#endif
```
Comment (juliantaylor) [inline scimath/Mathematics/ClassicalStatistics.tcc]: this is repeated often enough now that it should go into its own function like nThreadsMax

**Comment by juliantaylor** [comment]
I'm curious, is there a description on how the quantile computation works? it is not a trivial problem to solve on large data.

**Comment by dmehring** [comment]
Do you mean the tid bit (lines 1148-1152)?

There is probably not a good description of the quantile computation and I probably should write one.

**Comment by juliantaylor** [comment]
yes I meant the tid lines

**Comment by juliantaylor** [comment]
thanks, please just `commit --amend` these minor commit corrections in future

**Pull request merged**